code/datums/components/connect_range.dm

```diff
@@ -0,0 +1,91 @@
+/**
+ * This component behaves similar to connect_loc_behalf but for all turfs in range, hooking into a signal on each of them.
+ * Just like connect_loc_behalf, It can react to that signal on behalf of a seperate listener.
+ * Good for components, though it carries some overhead. Can't be an element as that may lead to bugs.
+ */
+/datum/component/connect_range
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+	var/atom/tracked
+
+	/// The component will hook into signals only on turfs not farther from tracked than this.
+	var/range
+	/// Whether the component works when the movable isn't directly located on a turf.
+	var/works_in_containers
+
+/datum/component/connect_range/Initialize(atom/tracked, list/connections, range, works_in_containers = TRUE)
+	if(!isatom(tracked) || isarea(tracked) || range <= 0)
+		return COMPONENT_INCOMPATIBLE
+	src.connections = connections
+	src.tracked = tracked
+	src.range = range
+	src.works_in_containers = works_in_containers
+
+/datum/component/connect_range/InheritComponent(datum/component/component, original, atom/movable/tracked, list/connections, range, works_in_containers)
+	// Not equivalent. Checks if they are not the same list via shallow comparison.
+	if(src.tracked != tracked || !compare_list(src.connections, connections))
+		return
+	if(src.range != range || src.works_in_containers != works_in_containers) //Update range and works_in_containers
+		src.range = range
+		src.works_in_containers = works_in_containers
+		update_signals(tracked, tracked.loc, TRUE)
+
+/datum/component/connect_range/RegisterWithParent()
+	RegisterSignal(tracked, COMSIG_MOVABLE_MOVED, .proc/on_moved)
+	RegisterSignal(tracked, COMSIG_PARENT_QDELETING, .proc/handle_tracked_qdel)
+	update_signals(tracked)
+
+/datum/component/connect_range/UnregisterFromParent()
+	unregister_signals(tracked)
+	UnregisterSignal(tracked, list(
+		COMSIG_MOVABLE_MOVED,
+		COMSIG_PARENT_QDELETING,
+	))
+
+	tracked = null
+
+/datum/component/connect_range/proc/handle_tracked_qdel()
+	SIGNAL_HANDLER
+	qdel(src)
+
+/datum/component/connect_range/proc/update_signals(atom/movable/movable, atom/old_loc, forced = FALSE)
```

Comment:
Could you please name movable to be more descriptive? really kinda messes up the rest of this rpoc